src/util/config.js

```diff
@@ -29,6 +33,22 @@ const config: Config = {
 
         return mapboxHTTPURLRegex;
     },
+    get API_TILEJSON_REGEX() {
+        // https://docs.mapbox.com/api/maps/mapbox-tiling-service/#retrieve-tilejson-metadata
+        return /^((https?:)?\/\/)?([^\/]+\.)?mapbox\.c(n|om)(\/v4\/.*\.json.*$)/i;
+    },
+    get API_SPRITE_REGEX() {
+        // https://docs.mapbox.com/api/maps/styles/#retrieve-a-sprite-image-or-json
+        return /^((https?:)?\/\/)?([^\/]+\.)?mapbox\.c(n|om)(\/styles\/v1\/)(.*\/sprite.*\..*$)/i;
+    },
+    get API_FONTS_REGEX() {
+        // https://docs.mapbox.com/api/maps/fonts/#retrieve-font-glyph-ranges
+        return /^((https?:)?\/\/)?([^\/]+\.)?mapbox\.c(n|om)(\/fonts\/v1\/)(.*\.pbf.*$)/i;
+    },
+    get API_STYLE_REGEX() {
+        // https://docs.mapbox.com/api/maps/styles/#retrieve-a-style
+        return /^((https?:)?\/\/)?([^\/]+\.)?mapbox\.c(n|om)(\/styles\/v1\/)(.*$)/i;
```

Comment:
Should we get rid of the version suffixes where it's possible? It might lead to false results if the suffixes change in the future.

Reply:
Yes good point, we can relax the suffix. The codebase has the version baked in a few other places so updating to a new style API version endpoint would probably require some special attention, but that would be one less place to think about it. I'll update that.

Reply:
Relaxed and updated in https://github.com/mapbox/mapbox-gl-js/pull/12343/commits/af215cf90ce754460ccfd84c86148f40b69569ce.